pepr/privacy/mia.py

```diff
@@ -0,0 +1,1182 @@
+    2. Train shadow models.
+    3. Generate attack model dataset.
+    4. Train attack models.
+    5. Evaluate attack models.
+
+    Parameters
+    ----------
+    attack_alias : str
+        Alias for a specific instantiation of the mia class.
+    attack_pars : dict
+        Dictionary containing all needed attack parameters:
+
+        * number_classes (int): Number of different classes the target model predicts.
+        * number_shadow_models (int): Number of shadow models to be trained.
+        * shadow_training_set_size (int): Size of the trainings set for each
+          shadow model. The corresponding evaluation sets will have the same size.
+        * create_compile_shadow_model (function): Function that returns a compiled
+          TensorFlow model (typically identical to the target model) used in the
+          training of the shadow models.
+        * create_compile_attack_model (function): Function that returns a compiled
+          TensorFlow model used for the attack models. The model output is expected to
+          be a single floating-point value per prediction.
+        * shadow_epochs (int): Number of training epochs of the shadow models.
+        * shadow_batch_size (int): Batch size used in the training of the
+          shadow models.
+        * attack_epochs (int): Number of training epochs of the attack models.
+        * attack_batch_size (int): Batch size used in the training of the
+          attack models.
+
+    data : numpy.ndarray
+        Dataset with all training samples used in the given pentesting setting.
+    labels : numpy.ndarray
+        Array of all labels used in the given pentesting setting.
+    data_conf: dict
+        Dictionary describing which record-indices are used to train the shadow
+        models, the target model(s) and which are used for the evaluation of the
+        attack.
+
+        * shadow_indices (list): List of indices describing which of the records
+          from data are used to train the shadow models.
+        * target_indices (list): List of indices describing which of the records
+          from data were used to train the target model(s).
+        * evaluation_indices (list): List of indices describing which of the records
+          from data are used to evaluate the attack.
+        * record_indices_per_target (numpy.ndarray): n*m array describing for all n
+          target models which m indices where used in the training.
+
+    target_models: iterable
+        List of target models which should be tested.
+
+    Attributes
+    ----------
+    attack_alias : str
+        Alias for a specific instantiation of the attack class.
+    attack_pars : dict
+        Dictionary containing all needed parameters fo the attack.
+    data : numpy.ndarray
+        Dataset with all training samples used in the given pentesting setting.
+    labels : numpy.ndarray
+        Array of all labels used in the given pentesting setting.
+    data_conf : dict
```

Comment:
Here we should add, how exactly the `data_conf` dictionary looks like.

Reply:
This is already explained in L63+ but there is no guarantee, that they are the same. Fixed in 03c76ef